src/shady-css-parser/parser.js

```diff
@@ -0,0 +1,331 @@
+import { matcher, tokenType } from './common';
+import { TokenCursor } from './token-cursor';
+
+function AtRule(name, parameters, block) {
+  return { type: 'at-rule', name, parameters, block };
+}
+
+function Comment(value) {
+  return { type: 'comment', value };
```

Comment:
Or, add an AstFactory 'interface' and you can leave out the node types altogether:

```
class AstFactory {
  atRule(name, parameters, block) {}
  comment(value) {}
  block(declarations) {}
  // etc...
}
```